sdk/core/azure-core/test/ut/extendable_enumeration_test.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/internal/extendable_enumeration.hpp>
+#include <gtest/gtest.h>
+#include <string>
+
+using namespace Azure::Core;
+
+class MyEnum : public Azure::Core::_internal::ExtendableEnumeration<MyEnum> {
+public:
+  MyEnum(std::string initialValue) : ExtendableEnumeration(std::move(initialValue)) {}
+  MyEnum() = default;
+
+  const static MyEnum Value1;
```

Comment:
I thought I saw `static const` elsewhere (which is the idiomatic ordering); again it's "only test code" so not a huge issue.

Reply:
There is no code which is "only test code".

At some point, I'll tell you the story of `RtlExtendedDivide32x64` and how what was a hack-o-rama routine intended to solve a theoretical but non-existent scenario ended up becoming a significant performance blocker in Windows Daytona (NT 3.5).